go/libraries/doltcore/sqle/index/dolt_index.go

```diff
@@ -258,17 +258,22 @@ RangeLoop:
 				}
 				cb.upperbound = val
 			}
+			if rangeColumnExpr.Type() == sql.RangeType_Null {
+				cb.boundsCase = boundsCase_isNull
+			}
 			rangeCheck[i] = cb
 		}
 
 		// If the suffix checks will always succeed (both bounds are infinity) then they can be removed to reduce the
-		// number of checks that are called per-row.
-		for i := len(rangeCheck) - 1; i >= 0; i-- {
+		// number of checks that are called per-row. Always leave one check to skip NULLs.
+		i := len(rangeCheck) - 1
+		for i > 0 && len(rangeCheck) > 1 {
```

Comment:
I'd rather have all of the variables in the `for` statement:
`for i := len(rangeCheck) - 1; i > 0 && len(rangeCheck) > 1; i-- {`